compare hash value when prove hashes are equal

darren0718 opened the pull request.

darren0718 committed:
1. implements checkpoint calculation by both accountDB and chainDB
2. get all groups by latest accountDB

darren0718 committed:
fork from chain-validation

darren0718 committed:
modify get history data

liuhaox1 committed:
crop develop

liuhaox1 committed:
memmax clear

liuhaox1 committed:
Merge branch 'features/get-history-data' into features/crop_trie

darren0718 committed:
implements prune command

darren0718 committed:
Merge branch 'features/crop_trie' into features/prune-full

darren0718 committed:
add fastcache

darren0718 committed:
finish full prune

darren0718 committed:
fix node

liuhaox1 committed:
crop2 dev

liuhaox1 committed:
bug fixed

liuhaox1 committed:
add shut down hook

liuhaox1 committed:
modify node cache

liuhaox1 committed:
modify queue

liuhaox1 committed:
remove no used import

liuhaox1 committed:
add trie buffer

liuhaox1 committed:
add cost log

liuhaox1 committed:
add cost log

liuhaox1 committed:
modify log show

liuhaox1 committed:
modify test

darren0718 committed:
implements fast replayer

liuhaox1 committed:
modify clear lock

darren0718 committed:
implements state verify

darren0718 committed:
Merge branch 'features/prune-full' of github.com:zvchain/zvchain into features/prune-full

liuhaox1 committed:
add log

liuhaox1 committed:
modify log

darren0718 committed:
add trie cache and meter

darren0718 committed:
Merge remote-tracking branch 'origin/features/prune-full' into features/prune-full

darren0718 committed:
Merge branch 'features/prune-full' into features/trie-verify-integrity

# Conflicts:
#	log/init.go

liuhaox1 committed:
add missing log

darren0718 committed:
verify integrity

darren0718 committed:
Merge branch 'develop' of github.com:zvchain/zvchain into features/trie-verify-integrity

liuhaox1 committed:
Merge branch 'develop' into features/trie-verify-integrity

liuhaox1 committed:
modify missing error

liuhaox1 committed:
Merge branch 'features/prune-full' into features/trie-verify-integrity

liuhaox1 committed:
Merge branch 'features/crop_trie2' into features/prune-full2

# Conflicts:
#	core/blockchain_store.go
#	storage/trie/database.go

liuhaox1 committed:
Merge branch 'features/trie-verify-integrity' into features/prune-full2

# Conflicts:
#	storage/trie/trie_test.go

liuhaox1 committed:
change config name

liuhaox1 committed:
remove no used

liuhaox1 committed:
modify

liuhaox1 committed:
bug fixed

liuhaox1 committed:
bug fixed

darren0718 committed:
prune full

liuhaox1 committed:
modify test case

liuhaox1 committed:
add test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
bug fixed

liuhaox1 committed:
add log

liuhaox1 committed:
disable rpc invoke prune mode

darren0718 committed:
prune full and verify

darren0718 committed:
Merge remote-tracking branch 'origin/features/trie-verify-integrity' into features/trie-verify-integrity

# Conflicts:
#	storage/trie/trie.go
#	storage/trie/trie_ext_test.go
#	storage/trie/trie_test.go

liuhaox1 committed:
add disable rpc

liuhaox1 committed:
Merge branch 'features/trie-verify-integrity' into features/prune-full2

# Conflicts:
#	storage/trie/trie_ext_test.go
#	storage/trie/trie_test.go

darren0718 committed:
prune full and verify

liuhaox1 committed:
modify test_case

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify output format

liuhaox1 committed:
modify test case

bird512 committed:
add checkHash

bird512 committed:
Merge branch 'features/prune-full2' of github.com:zvchain/zvchain into features/prune-full2

liuhaox1 committed:
modify test case

liuhaox1 committed:
Merge branch 'features/prune-full2' of https://github.com/zvchain/zvchain into features/prune-full2

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

darren0718 committed:
database meter out with time

darren0718 committed:
log error when reading from disk

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

darren0718 committed:
fix some problem

darren0718 committed:
persist state cache on Close

darren0718 committed:
persist state cache on Close

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify name

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

darren0718 committed:
Merge branch 'features/trie-verify-integrity' into features/prune-full2

# Conflicts:
#	cmd/gzv/cli/gzv.go
#	core/blockchain.go
#	core/state_prune.go
#	storage/account/accountdb.go

darren0718 committed:
Merge remote-tracking branch 'origin/features/prune-full2' into features/prune-full2

# Conflicts:
#	core/blockchain.go

liuhaox1 committed:
modify shutdown hook

liuhaox1 committed:
Merge branch 'features/prune-full2' of https://github.com/zvchain/zvchain into features/prune-full2

darren0718 committed:
update test

liuhaox1 committed:
modify field name

liuhaox1 committed:
Merge branch 'features/prune-full2' of https://github.com/zvchain/zvchain into features/prune-full2

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify test case

liuhaox1 committed:
add nil check

liuhaox1 committed:
modify method name

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify field name

liuhaox1 committed:
add cost log

liuhaox1 committed:
modify test case

liuhaox1 committed:
modify pop to peek

darren0718 committed:
update test

darren0718 committed:
Merge branch 'features/prune-full2' of github.com:zvchain/zvchain into features/prune-full2

liuhaox1 committed:
reset nil develop

liuhaox1 committed:
Merge branch 'features/prune-full2' of https://github.com/zvchain/zvchain into features/prune-full2

liuhaox1 committed:
modify method name

ysh0566 committed:
ctvm version fix

darren0718 committed:
fix checkpoint calclation when first scan meets missing node

darren0718 committed:
Merge remote-tracking branch 'origin/features/prune-full2' into features/prune-full2

liuhaox1 committed:
bug fixed

liuhaox1 committed:
modify log

liuhaox1 committed:
modify log

liuhaox1 committed:
check add

liuhaox1 committed:
modify test  case

darren0718 committed:
compare hash value when prove hashes are equal

darren0718 committed:
compare hash value when prove hashes are equal

liuhaox1 committed:
modify test case

sesamegu reviewed (changes requested):
need some test cases

liuhaox1 committed:
check small db is nil

liuhaox1 committed:
triegc not empty check

darren0718 committed:
finish traverse

liuhaox1 committed:
optimization

sesamegu committed:
refactor

liuhaox1 committed:
add notes

darren0718 committed:
1. add pprof
2. traverse only once for same root
3. only traverse the groupkey for group account, others are pruned

liuhaox1 committed:
Merge branch 'features/prune-full2' of https://github.com/zvchain/zvchain into features/prune-full2

lavrock committed:
revert subproject version

liuhaox1 committed:
rpc bug fix

liuhaox1 committed:
Merge branch 'features/prune-full2' of https://github.com/zvchain/zvchain into features/prune-full2

darren0718 committed:
use ticker instead of time.Since which is cpu intense

liuhaox1 committed:
param check

liuhaox1 committed:
modify config

liuhaox1 committed:
check iter seek

liuhaox1 committed:
merge modify

liuhaox1 committed:
remove no use code

darren0718 committed:
fix checkpoint Logger nil

darren0718 committed:
Merge branch 'features/prune-full2' into features/prune-full-opti

# Conflicts:
#	storage/trie/database.go

darren0718 committed:
CPU cores and max open files can be set

liuhaox1 committed:
modify small db params

liuhaox1 committed:
Merge branch 'features/prune-full2' of https://github.com/zvchain/zvchain into features/prune-full2

sesamegu committed:
refactor

darren0718 committed:
prune test

darren0718 committed:
Merge remote-tracking branch 'origin/features/prune-full2' into features/prune-full2

darren0718 committed:
add test for group data pruned

liuhaox1 committed:
cancel db deleted  compensation scheme

liuhaox1 committed:
Merge branch 'features/prune-full2' of https://github.com/zvchain/zvchain into features/prune-full2

# Conflicts:
#	core/blockchain_action.go

liuhaox1 committed:
modify test case

liuhaox1 committed:
put checkpoint method to prune mode

darren0718 committed:
add checkpoint compare test

darren0718 committed:
add checkpoint compare test

liuhaox1 committed:
use bytes copy instead rlp

liuhaox1 committed:
Merge branch 'features/prune-full2' of https://github.com/zvchain/zvchain into features/prune-full2

liuhaox1 committed:
reset near modify

liuhaox1 committed:
damaged print console

liuhaox1 committed:
modify panic to error

liuhaox1 committed:
modify default to 512

liuhaox1 committed:
set default param

darren0718 committed:
fix unit test

darren0718 committed:
Merge remote-tracking branch 'origin/features/prune-full2' into features/prune-full2

liuhaox1 committed:
modify test case

liuhaox1 committed:
Merge branch 'features/prune-full2' of https://github.com/zvchain/zvchain into features/prune-full2

darren0718 committed:
Merge pull request #454 from zvchain/features/prune-full2

Features/prune full2

liuhaox1 committed:
modify rpc notice

sesamegu committed:
Merge pull request #463 from zvchain/bugfix/prune_rpc_notice

modify rpc notice

liuhaox1 committed:
recode

liuhaox1 committed:
key copy modify

liuhaox1 committed:
key copy fix

liuhaox1 committed:
close persistent delete small db

liuhaox1 committed:
recode

liuhaox1 committed:
modify if

liuhaox1 committed:
remove root key

liuhaox1 committed:
remove cp log

liuhaox1 committed:
modify key

liuhaox1 committed:
add log

liuhaox1 committed:
order fix

darren0718 committed:
refactor codes

darren0718 committed:
Merge remote-tracking branch 'origin/features/prune_mode_recode' into features/prune_mode_recode

# Conflicts:
#	core/blockchain_action.go
#	core/trie_small_db_store.go

liuhaox1 committed:
bug fix

liuhaox1 committed:
bug fix

darren0718 committed:
rename parseHeight

liuhaox1 committed:
add log

darren0718 committed:
fix iter key not contains prefix

darren0718 committed:
Merge remote-tracking branch 'origin/features/prune_mode_recode' into features/prune_mode_recode

darren0718 committed:
move merge to big db

darren0718 committed:
modify mergeSmallDbDataToBigDB to repairStateDatabase

darren0718 committed:
fix lastBH

liuhaox1 committed:
add test

liuhaox1 committed:
bug fixed

darren0718 committed:
don't persist fastcache

darren0718 committed:
Merge remote-tracking branch 'origin/features/prune_mode_recode' into features/prune_mode_recode

sesamegu committed:
Merge pull request #466 from zvchain/features/prune_mode_recode

Features/prune mode recode

liuhaox1 committed:
reset bug fix

liuhaox1 committed:
reset

liuhaox1 committed:
refix

darren0718 committed:
double check when repair database

liuhaox1 committed:
modify log

liuhaox1 committed:
add error check

liuhaox1 committed:
clear double check

liuhaox1 committed:
remove no used

liuhaox1 committed:
modify log

liuhaox1 committed:
modify log

ysh0566 committed:
Features/add license (#468)

* add license

liuhaox1 committed:
fix small db lose one height state data

liuhaox1 committed:
reset modify

liuhaox1 committed:
merge for

liuhaox1 committed:
cp nil fix

liuhaox1 committed:
cp nil fix

liuhaox1 committed:
add test

liuhaox1 committed:
cp nil fix

sesamegu committed:
Merge pull request #471 from zvchain/bugfix/cp_nil_fix

cp nil fix

lavrock committed:
modify default max broadcast count

sesamegu committed:
Merge pull request #472 from zvchain/orgin/features/max_broadcast_count

modify default max broadcast count

darren0718 committed:
add zip3: solve weight comparison when block proves are equal

darren0718 committed:
Merge remote-tracking branch 'origin/features/same-pv' into features/same-pv

# Conflicts:
#	middleware/types/core.go

jyiigpgf committed:
* update vm submodule
* VM support ZIP003
